open_spiel/games/CMakeLists.txt

```diff
@@ -1,6 +1,10 @@
 set(GAME_SOURCES
   backgammon.cc
   backgammon.h
+  battleship.cc
+  battleship.h
+  battleship_types.h
+  battleship_types.cc
```

Comment:
Can you add battleship_test.cc with some basic tests, and add the test to this file as well?

It's ok to just have the simulation tests for now, they do a number of checks from the core API. 

You can run some examples using example.cc and manually verify the boards.

Reply:
Yes, I will shortly! I have first fixed a pretty annoying memory bug that was caught by `asan` about taking `absl::string_view`s to temporaries.

Reply:
I think I spoke too early... I hit a couple of issues related to `GameParameter`s. I will open issue and work towards fixing those as required steps for this PR.

Reply:
I pretty much never use `string_view` because it is easy to run into non-trivial bugs like that and IMO what it brings to the table is not worth that cost.

That other issue I will address in the other thread.